src/CGALCache.h

```diff
@@ -12,9 +12,9 @@ class CGALCache
 
 	static CGALCache *instance() { if (!inst) inst = new CGALCache; return inst; }
 
-	bool contains(const std::string &id) const { return this->cache.contains(id); }
-	shared_ptr<const class CGAL_Nef_polyhedron> get(const std::string &id) const;
-	bool insert(const std::string &id, const shared_ptr<const CGAL_Nef_polyhedron> &N);
+	bool contains(const size_t id) const { return this->cache.contains(id); }
```

Comment:
I find this slightly scary. Hashed containers are usually designed such that they gracefully handle hash collisions. In this case, it looks like we'd potentially do significant damage in case of a hash collision.

Reply:
Hmm, good point, it looks like std::hash I'm using in `Tree.cc` is not particularly good about collisions https://stackoverflow.com/questions/7968674/unexpected-collision-with-stdhash
especially because according to that link its not required to hash the entire string.
Do you think a different hash algorithm exists that is robust enough to avoid collision on its own or it should be handled explicitly in some other way.   I'm not sure the best approach

Reply:
I guess CGALCache is not going to be storing anywhere near **all** the node keys, so maybe its best that I just revert to using getIdString as before.

Reply:
The CSGTextCache on the other hand does keep all node, I think.  So if I went back to strings, I still feel that should be rewritten.

Reply:
Just to follow up on this even though it seems fixed now: Using hashes as keys is always a bad idea. The key needs to be unique. This is what hash maps do; store all items with the same hash in the same bucket and reverting to full key comparison only for hash collisions.
If you're worried about the time taken to compute the hash from a key, the solution is probably to lazily compute the hash and cache it. ...in case the same object is processed many times during a traversal.